web/controllers/resources.ex

```diff
@@ -0,0 +1,76 @@
+defmodule App.Resources do
+  @moduledoc """
+  # Functions for querying CMS Database for resources
+  """
+
+  use App.Web, :controller
+
+  alias App.CMSRepo
+
+  def get_all_resources(type) do
+    query = from r in "#{type}s_#{type}page",
+      select: %{
+        id: r.page_ptr_id,
+        heading: r.heading,
+        url: r.resource_url,
+        body: r.body,
+        priority: r.priority
+      }
+
+      query
+      |> CMSRepo.all
+      |> Enum.map(&(Map.merge(&1, %{type: "#{type}s"})))
+      |> Enum.map(&(get_all_tags(&1)))
+      |> Enum.sort(&(&1[:priority] <= &2[:priority]))
+  end
+
+  def create_tag_query(tag, type) do
+    query = from t in "taggit_tag",
+      where: t.name == ^tag,
+      join: cat in ^"#{type}_categorytag",
+      where: cat.tag_id == t.id,
+      join: a in ^"#{type}_#{String.slice(type, 0..-2)}page",
```

Comment:
maybe change the `type` that we are passing down to be `"resource"` rather than `"resources"`